Utilities/Thread.cpp

```diff
@@ -2019,20 +2019,95 @@ u16 thread_ctrl::get_affinity_mask(thread_class group)
 		}
 		case native_core_arrangement::amd_ccx:
 		{
-			u16 spu_mask, ppu_mask, rsx_mask;
-			if (thread_count >= 16)
+			u32 spu_mask, ppu_mask, rsx_mask;
+			const auto system_id = utils::get_system_info();
+			if (thread_count >= 32)
 			{
-				// Threadripper, R7
-				// Assign threads 8-16
-				// It appears some windows code is bound to lower core addresses, binding 8-16 is alot faster than 0-7
-				ppu_mask = spu_mask = 0b1111111100000000;
-				rsx_mask = all_cores_mask;
+				if (system_id.find("3950X") != std::string::npos)
+				{
+					// zen2
+					// Ryzen 9 3950X
+					// Assign threads 9-30
+					ppu_mask = 0b1111000000000000000000000000;
+					spu_mask = 0b111111110000000000000000;
+					rsx_mask = 0b11111100000000;
+				}
+				else if (system_id.find("2970WX") != std::string::npos)
+				{
+					// zen+
+					// Threadripper 2970WX
+					// Assign threads 9-24
+					ppu_mask = spu_mask = 0b111111000000000000;
```

Comment:
```suggestion
					ppu_mask = 0b111111000000000000
					spu_mask = ppu_mask;
```
ditto for the rest

Reply:
are you sur ? is like this on master, if @kd-11 can said me if is good like that i set ^^

Reply:
just preference i guess, leave it if kd said so

Reply:
yues, i set it like that actually, is beter clear for read ^^